indexing-solr/src/main/java/edu/unc/lib/boxc/indexing/solr/filter/SetDatastreamFilter.java

```diff
@@ -128,29 +128,125 @@ private String getExtent(List<BinaryObject> binList) {
+                if (audioMd != null) {
+                    var audioTime = formatTime(audioMd);
+                    return "xx" + audioTime;
+                }
             }
-            return extent;
+
+            return null;
         } catch (RepositoryException | FedoraException e) {
             log.warn("Unable to parse FITS for {}", fitsId, e);
             return null;
         }
     }
 
+    private String formatVideoExtent(List<Element> trackInfo, String pid) {
+        var numTracks = trackInfo.size();
+        var videoTrack = 0;
+        var audioTrack = 0;
+
+        if (numTracks > 1) {
+            for (int i = 0; i < numTracks; i++) {
+                var type = trackInfo.get(i).getAttributeValue("type");
+                if (type.equals("video")) {
+                    videoTrack = i;
+                }
+                if (type.equals("audio")) {
+                    audioTrack = i;
+                }
+            }
+        }
+
+        var videoInfo = trackInfo.get(videoTrack);
+        var videoTime = formatTime(videoInfo);
+        var videoHeight = videoInfo.getChildTextTrim("width", FITS_NS);
+        var videoWidth = videoInfo.getChildTextTrim("height", FITS_NS);
+
+        // Some videos have separate tracks for the audio content
+        String audioTime = null;
+        if (numTracks > 1) {
+            audioTime = formatTime(trackInfo.get(audioTrack));
+        }
+
+        // Audio and video don't necessarily have the same play times, so select the larger one if both are present
+        String trackTime = (audioTime == null || Integer.parseInt(videoTime) >= Integer.parseInt(audioTime))
+                ? videoTime : audioTime;
+        var extent = formatExtent(videoHeight, videoWidth, pid);
+        return (extent == null) ? "xx" + trackTime : extent + "x" + trackTime;
+    }
+
+    private String formatTime(Element durationElement) {
+        var durationMilliseconds = durationElement.getChild("milliseconds", FITS_NS);
+        var duration  = durationElement.getChild("duration", FITS_NS);
+
+        if (durationMilliseconds != null) {
+            return normalizeTime(durationMilliseconds.getTextTrim());
+        }
+
+        if (duration != null) {
+            return normalizeTime(duration.getTextTrim());
+        }
+
+        return "-1";
+    }
+
+    private String normalizeTime(String duration) {
+        String output = "";
+        Pattern pattern = Pattern.compile("\\d+:\\d+:\\d+", Pattern.CASE_INSENSITIVE);
+        Matcher matcher = pattern.matcher(duration);
+        boolean matchFound = matcher.find();
+
+        if (matchFound) {
+            var durationParts = duration.split(":");
+            var hoursToSeconds = Integer.parseInt(durationParts[0]) * 60 * 60;
+            var minutesToSeconds = Integer.parseInt(durationParts[1]) * 60 ;
+            var seconds = Integer.parseInt(durationParts[2]);
+
+            output += hoursToSeconds + minutesToSeconds + seconds;
```

Comment:
Seems like output only ever gets one value put into it, so we can skip assigning it to `""` and assign it here rather than doing `+=`